### ISSUE TITLE
completion UI

### ISSUE THREAD

**Issue by neurocyte** [neurocyte/flow#86]

**Comment by plyght** [comment]
@neurocyte Will this include a completion UI for opening files here? (see image below):

<img width="853" alt="Image" src="https://github.com/user-attachments/assets/e0d85f7c-deea-4d25-85aa-12bd0a1309e6" />

**Comment by plyght** [comment]
Theoretically, you could use the same thing as this: 

<img width="853" alt="Image" src="https://github.com/user-attachments/assets/76c1e6e9-9439-4eca-b5d4-cd9bc80b9754" />

**Comment by neurocyte** [comment]
> [@neurocyte](https://github.com/neurocyte) Will this include a completion UI for opening files here? (see image below):

No, this would be the file browser UI. The file browser already has completion, but I guess you're looking for a UI which has popup lists of files. This is kind of planned (along with breadcrumb navigation), but there is no feature request for it yet.

**Comment by plutonium-239** [comment]
Just to get an idea, what kind of UI are we thinking here?

There are broadly 3 types (if you know of any more please add to this)

<details><summary>Here are some inspirations:</summary>

## Details List style
- mini.nvim
![Image](https://github.com/user-attachments/assets/f9cdefd4-595f-40a0-8be2-1c5872f449c5)
- ptpython
![Image](https://github.com/user-attachments/assets/3c458ee2-b6aa-4516-8237-5214e7cd7634)
- sublime
![Image](https://github.com/user-attachments/assets/86794cbb-3d7d-4755-99b2-60f33ceacfd7)

## Simple Grid style
- ipython 
![Image](https://github.com/user-attachments/assets/9e5d715d-8b5c-461f-b117-999b2b73a870)

## Simple Row style
- micro (shows up in the status line)
![Image](https://github.com/user-attachments/assets/e3f2f352-ab50-4fae-9a2d-9d436b1208cf)

</details> 

In general a detailed list is preferred by most, a grid is unwieldy and doesn't give enough information, and completions in the status line seems like an afterthought (micro does not have any modals, although it does have panels and a plugin that says it shows completions in panels but I could not get it to work reliably).

**Comment by neurocyte** [comment]
A detail list with type symbol seems most appealing to me. Perhaps with the option to display it inline at the cursor or in a (vertical) side bar.

**Comment by zai1208** [comment]
would this be implemented using `src/tui/info_view.zig`?

**Comment by neurocyte** [comment]
> would this be implemented using `src/tui/info_view.zig`?

For the extended item info, yes, probably.

**Comment by neurocyte** [comment]
I going to close this as I think we've reached the MVP for the v0.6.0 release. Improvements can be put in new feature requests.
